pkg/profile/artifact.go

```diff
@@ -156,36 +177,50 @@ func (p *Profile) MakeArtifacts() ([]runtime.Object, error) {
 	return objs, nil
 }
 
-func (p *Profile) makeArtifacts() (*sourcev1.GitRepository, []*helmv2.HelmRelease, []*kustomizev1.Kustomization, error) {
+func (p *Profile) makeArtifacts() (*sourcev1.GitRepository, []*sourcev1.HelmRepository, []*helmv2.HelmRelease, []*kustomizev1.Kustomization, error) {
 	var helmResources []*helmv2.HelmRelease
+	var helmReps []*sourcev1.HelmRepository
 	var kustomizeResources []*kustomizev1.Kustomization
+	var gitRes *sourcev1.GitRepository
 
 	for _, artifact := range p.definition.Spec.Artifacts {
 		switch artifact.Kind {
-		case "HelmChart":
+		case profilesv1.HelmChartKind:
 			helmRes, err := p.makeHelmRelease(artifact)
 			if err != nil {
-				return nil, nil, nil, fmt.Errorf("failed to create HelmRelease resource: %w", err)
+				return nil, nil, nil, nil, fmt.Errorf("failed to create HelmRelease resource: %w", err)
 
 			}
 			helmResources = append(helmResources, helmRes)
-		case "Kustomize":
+			if artifact.Path != "" {
+				gitRes, err = p.makeGitRepository()
+				if err != nil {
+					return nil, nil, nil, nil, fmt.Errorf("failed to create GitRepository resource: %w", err)
+				}
```

Comment:
```suggestion
			if artifact.Path != "" && gitRes == nil {
				gitRes, err = p.makeGitRepository()
				if err != nil {
					return nil, nil, nil, nil, fmt.Errorf("failed to create GitRepository resource: %w", err)
				}
```

No need to keep recreating this each time :smile:

Reply:
Ah, good point! Thanks :)